wikimedia/wikipedia-ios
# Apple Pay - finalizing authorization sheet and thank you toast presentation

**Phabricator:** 
https://phabricator.wikimedia.org/T345852
https://phabricator.wikimedia.org/T288285

### Notes
This PR represents the last pieces needed to get the Donate form to display the payment sheet and thank you toast once the `submitPayment` call succeeds. I expect smaller followup PRs will be needed once the endpoints are finalized.





This PR is dependent on https://github.com/wikimedia/wikipedia-ios/pull/4619.

### Test Steps

Prep work: On a test device, log out of your iCloud account in device settings, and log into a sandbox testing account (see iOS Sandbox Tester details in 1Password). Do not enable 2-factor if prompted. Once logged into sandbox account, go to the Apple Wallet app and add a [sandbox testing card](https://developer.apple.com/apple-pay/sandbox-testing/). (I have had luck with the top Discover one).

To test Xcode Cloud script injection:
1. Download the ad-hoc build from Staging build number 43 in App Store Connect and drag it to a connected device on your machine. 
2. Confirm you can successfully go through steps 3 - 5 below

To test locally:
1. From repo root directory, run this command in terminal: `./scripts/copy_staging_environment_vars.sh "{merchantID}" "ABCDPaymentAPIKeyEFGH"` replacing {merchantID} with our test merchant ID. Reach out to me if you need help finding this. This payment API key is fake, but we aren't using it in the endpoint calls yet so this is just to confirm it gets copied to the Info.plist file properly.
2. After running, confirm you see a change in `Staging-Info.plist` (added MerchantID and PaymentsAPIKey) and `Wikipedia.entitlements` (added merchant ID to `com.apple.developer.in-app-payments` key) in git. 

3. Launch app. Background, foreground, and pull to refresh Explore feed to trigger config fetches.
4. Go through the form and tap the Apple Pay button. Confirm you now see a payment authorization sheet. Confirm you are prompted to enter billing address, name, and email details if needed, and can complete a payment.
5. Confirm after completing payment, you are taken back to the Settings root and see a thank you toast. 

Note: Just an FYI. For the Staging app, we are mocking the donate endpoint calls, thanks to a [temporary commit](https://github.com/wikimedia/wikipedia-ios/pull/4619/commits/b3aea1bdc197f820c6adf8e7bf5f24ece9467053) in the first PR. This mocking commit is reverted in the next PR (#4645)
